DNN Platform/Modules/Journal/ServicesController.cs

```diff
@@ -119,14 +119,31 @@ public HttpResponseMessage Create(CreateDTO postData)
                     postData.ProfileId = UserInfo.UserID;
                 }
 
+                if (postData.ProfileId != UserInfo.UserID)
+                {
+                    var profileUser = UserController.Instance.GetUser(PortalSettings.PortalId, postData.ProfileId);
+                    if (profileUser == null || (!UserInfo.IsInRole(PortalSettings.AdministratorRoleName) && !Utilities.AreFriends(profileUser, UserInfo)))
+                    {
+                        throw new ArgumentException("you have no permission to post journal on current profile page");
+                    }
+                }
+
                 if (postData.GroupId > 0)
                 {
                     postData.ProfileId = -1;
```

Comment:
I know this is an old line of code, but why is it is needed? don't we keep the UserID when posting to a group?